### DISCUSSION TITLE
Codeforces: Bitki

### DISCUSSION THREAD

**Discussion by OmerAkbayrak** [sonsuzus/sonsuzus#506] [Soru Cevap] [answered]
### **BİR BİTKİ**
Cüceler çok ilgi çekici bir bitki ektiler, hangi üçgen "yukarı" bakmakta. Bu bitki çok eğlenceli bir özelliğe sahip. Bir yıl sonra üçgen bitkisindeki "yukarı" bakan üçgenler dört üçgen bitkisine ayrılır: Bunlardan üçü "yukarıya" bakacak ve bir tanesi de "aşağıya" bakacak. Bir yıl daha sonra, her bir üçgen bitkisi dört üçgen bitkisine ayrılır: Bunlardan üçü ana bitkiyle aynı yöne bakar, ve bunlardan biri ters yöne bakar. Sonra her yıl bu işlem tekrarlanır. Aşağıdaki figür de bu işlemi göstermektedir.

![56](https://github.com/user-attachments/assets/c799e6cd-3f7d-4539-b32f-f173cd303424)

Cücelere n yıl içerisinde kaç tane üçgen bitkisinin "yukarı" bakacağını bulmalarında yardım et.

**Girdi**
İlk satır tek bir n sayısı içerir.(0<n<10¹⁸) - Bu bitkinin büyüyeceği tüm yılların sayısı.

> _Lütfen %11d belirleyicisini veya C++' daki 64 bit sayıları kullanmayın. cin, cout yayınları veya %I64d belirleyicisi önerilir_.

Çıktı
Tek bir sayı yazdırın - n yıl sonra "yukarı" bakacak olan bitki sayısının 1000000007 'ye (10⁹ + 7) bölümündeki kalanı gösterecek.

**Örnekler:**

> **Girdi**
>  1

> **Çıktı**
>  3

__

> **Girdi**
> 2

> **Çıktı**
> 10
**Not:**
Yukarıdaki örneklerdeki ilk sayı yukarıdaki figürlerden ikinci üçgeni gösterir. Üçüncü figür ise ikinci örneği temsil eder.

[https://codeforces.com/problemset/problem/185/A](url)
Ömer Akbayrak

**Comment by OmerAkbayrak** [comment, marked as answer]
```python
n=int(input("n sayısını giriniz: "))
def asagisay(n):
    if n==0:
        return 0
    elif n==1:
        return 1
    else:
        return 3*asagisay(n-1)+yukarisay(n-1)

def yukarisay(n):
    if n==0:
        return 1
    elif n==1:
        return 3
    else:
        return 3*yukarisay(n-1)+asagisay(n-1)

print  (yukarisay(n%1000000007))
    
    

```

**Reply by sonsuzus** [reply]
Biraz yapay zeka havası var ama :)

**Reply by OmerAkbayrak** [reply]
Yok hocam soruyu çeviriyi zaten derste yapmıştım.  soruyu da şimdi kendim çözdüm.

**Reply by sonsuzus** [reply]
Harika o zaman. Olmuşsun sen.

**Reply by OmerAkbayrak** [reply]
Teşekkür ederim hocam.
